extensions/bittrex/exchange.js

```diff
@@ -0,0 +1,289 @@
+var bittrex_authed = require('node.bittrex.api'),
+    bittrex_public = require('node.bittrex.api'),
+    path = require('path'),
+    moment = require('moment'),
+    n = require('numbro'),
+    colors = require('colors')
+
+
+/**
+ ** Bittrex API
+ ** 
+ ** please note: unfortunately getmarkethistory does not work based on time
+ ** this means that we cannot do any backfilles, but paper trading should be fine
+ ** https://github.com/n0mad01/node.bittrex.api/issues/17
+ **
+ **/
+module.exports = function container(get, set, clear) {
+  var c = get('conf')
+  var recoverableErrors = new RegExp(/(ESOCKETTIMEDOUT|ETIMEDOUT|ECONNRESET|ECONNREFUSED|ENOTFOUND|Invalid nonce|Rate limit exceeded)/)
+  var shownWarning = false
+
+  bittrex_authed.options({
+    'apikey' : c.bittrex.key.trim(),
+    'apisecret' : c.bittrex.secret.trim()
+  })
+
+  function joinProduct(product_id) {
+    return product_id.split('-')[0] + '-' + product_id.split('-')[1]
+  }
+
+  function retry(method, args, error) {
+    if (error.message.match(/Rate limit exceeded/)) {
```

Comment:
`API:Rate limit exceeded` is Kraken specific, I picked this generic message in case they decide to rate limit in the future (as they should, really).